cogboard-app/src/main/kotlin/com/cognifide/cogboard/widget/type/JiraBucketWidget.kt

```diff
@@ -0,0 +1,58 @@
+package com.cognifide.cogboard.widget.type
+
+import com.cognifide.cogboard.widget.AsyncWidget
+import io.vertx.core.Vertx
+import io.vertx.core.json.JsonArray
+import io.vertx.core.json.JsonObject
+import com.cognifide.cogboard.CogboardConstants as CC
+
+class JiraBucketWidget(vertx: Vertx, config: JsonObject) : AsyncWidget(vertx, config) {
+
+    private val bucketQueries: JsonArray = config.getJsonArray("bucketQueries")
+
+    private val buckets: JsonArray = prepareBuckets()
+
+    private fun prepareBuckets(): JsonArray {
+        val preparedBuckets = JsonArray()
+        bucketQueries.forEach {
+            if (it is JsonObject) {
+                val bucket = JsonObject()
+                bucket.put("id", it.getString("id"))
+                bucket.put("name", it.getString("bucketName"))
+                preparedBuckets.add(bucket)
+            }
+        }
+        return preparedBuckets
+    }
+
+    override fun handleResponse(responseBody: JsonObject) {
+        val response = JsonObject()
+        val issues = responseBody.getJsonArray("issues").size()
+        val bucketId = responseBody.getString("requestId")
+
+        buckets.forEach {
+            val key = (it as JsonObject).getString("id")
+            if (bucketId == key) {
+                it.put("issueCounts", issues)
+            }
+        }
+        response.put("buckets", buckets)
+        send(JsonObject()
+                .put(CC.PROP_CONTENT, response))
```

Comment:
you can create a response object here directly if you want

Reply:
I think it looks for CC.PROP_CONTENT. It doesn't work when done like that:
`send(response)`

Reply:
```
send(
    JsonObject().put(CC.PROP_CONTENT, 
        JsonObject().put("buckets", buckets)))
```

Reply:
Changed